pkg/vere/main.c

```diff
@@ -1215,29 +1223,63 @@ _cw_eval(c3_i argc, c3_c* argv[])
       exit(1);
     }
   }
-
-  printf("eval:\n");
-
-  //  +wish for an eval gate (virtualized twice for pretty-printing)
-  //
-  u3_noun gat = u3v_wish("|=(a=@t (sell (slap !>(+>.$) (rain /eval a))))");
-  u3_noun res;
-  {
+  fprintf(stderr, "eval:\n");
+  if ( c3n == jam_l ) {
+    //  +wish for an eval gate (virtualized twice for pretty-printing)
+    //
+    u3_noun gat = u3v_wish("|=(a=@t (sell (slap !>(+>.$) (rain /eval a))))");
+    u3_noun res;
+    {
+      u3_noun sam = u3i_string(evl_c);
+      u3_noun cor = u3nc(u3k(u3h(gat)), u3nc(sam, u3k(u3t(u3t(gat)))));
+      res = u3m_soft(0, u3n_kick_on, cor);
+    }
+    if ( 0 == u3h(res) ) {  //  successful execution, print output
+      u3_pier_tank(0, 0, u3k(u3t(res)));
+    }
+    else {                  //  error, print stack trace
+       u3_pier_punt_goof("eval", u3k(res));
+    }
+    u3z(res);
+    u3z(gat);
+  } else {
     u3_noun sam = u3i_string(evl_c);
-    u3_noun cor = u3nc(u3k(u3h(gat)), u3nc(sam, u3k(u3t(u3t(gat)))));
-    res = u3m_soft(0, u3n_kick_on, cor);
-  }
-
-
-  if ( 0 == u3h(res) ) {  //  successful execution, print output
-     u3_pier_tank(0, 0, u3k(u3t(res)));
-  }
-  else {                  //  error, print stack trace
-     u3_pier_punt_goof("eval", u3k(res));
+    u3_noun res = u3v_wish_n(sam);
+    c3_d bits = 0;
+    c3_d len_d = 0;
+    c3_y* byt_y;
+    bits = u3s_jam_xeno(res, &len_d, &byt_y);
+    if ( c3n == khan_l ) {
+      fprintf(stderr,"jammed noun: ");
+      for ( size_t p=0; p < len_d; p++ ){
+        fprintf(stdout,"\\x%2x", byt_y[p++]);
+      }
+      fprintf(stderr,"\n");
+    } else {
+        fprintf(stderr,"khan jammed noun: ");
+        c3_y out_y[5];
+        out_y[0] = 0x0;
+        out_y[1] = ( len_d        & 0xff);
+        out_y[2] = ((len_d >>  8) & 0xff);
+        out_y[3] = ((len_d >> 16) & 0xff);
+        out_y[4] = ((len_d >> 24) & 0xff);
+        fwrite(out_y, 1, 5, stdout);
+        if( ferror(stdout))
+        {
+           fprintf(stderr, "Write Failed : %s\n",strerror(errno) );
+           exit(1);
+        }
+        fwrite(byt_y, 1, len_d, stdout);
+        if( ferror(stdout))
+        {
+           fprintf(stderr, "Write Failed : %s\n",strerror(errno) );
+           exit(1);
+        }
+        fprintf(stderr, "\n");
+      }
```

Comment:
The indentation in this part is all kinds of messed up:
- body indented by 4 instead of 2
- bodies of `if` statements indented by 3

Also, it's full of inconsistent style, with regards to space before/after conditionals, arguments, and parentheses.

Reply:
I looked through the commit and tried to fix up everything. Is there a formatting guide somewhere I'm not aware of?

Reply:
There are some conventions in [here](https://github.com/urbit/urbit/blob/master/doc/spec/u3.md). There has been murmur about introducing auto-formatting using some third-party tool, though this was months ago when Tlon was managing the repo. New community ownership might be more interested if it's reintroduced.